.gitignore

```diff
@@ -6,6 +6,8 @@ delivery-service/ds/target
 **/Cargo.lock
 *.code-workspace
 /test_vectors/
+fuzz/corpus
+fuzz/artifacts
```

Comment:
I disagree on this one. `.cargo/config.toml` should be sourced since it can contain command aliases as well as compiler configuration.
I think local patches should go in root `Cargo.toml` and changed to a remote branch when pushed in order for the CI to work

Reply:
That's not how things work.
`.cargo` is not supposed on version control. If you don't want to this, we won't have a workspace in this repository.

Reply:
I added .cargo to gitignore

Reply:
> .cargo is not supposed on version control. If you don't want to this, we won't have a workspace in this repository.

Some things to keep in mind: 
* The [docs](https://doc.rust-lang.org/cargo/reference/config.html#hierarchical-structure) mention that .cargo *can* be checked in source control
* From my experience, there are many targets that simply do not compile and/or link without the proper `.cargo/config.toml` configuration. For example, Android targets typically silently fail compiling (i.e. rustc returns 0 but no library is produced) because they require a valid Android NDK to be set as the linker. If you want reproducible builds while targeting "exotic" platforms, you will need to check those files into source control.
* The `[patch]` section is a part of the [Cargo manifest format](https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section), not the [configuration format](https://doc.rust-lang.org/cargo/reference/config.html#configuration-format). You **cannot** patch dependencies using `.cargo/config.toml`. 

Otherwise I agree with you that it's *usually* not needed and creates more places to look at configuration, making it unnecessarily complicated :(

Reply:
Fair point. I did not know you can patch this. It is limited though compared to `[patch]` or as the [documentation](https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#paths-overrides) says `As a result usage of a path override is typically isolated to quick bug fixes rather than larger changes`.  

Anyway I'm not opinionated on this, it's also fine not to source `.cargo`. We can move on.